ledger/src/ui/src/bolos_ux.c

```diff
@@ -326,86 +320,15 @@ static void sample_main(void) {
-                    nvm_write(
-                        (void *)PIC(N_onboarded_ui), (void *)&aux, sizeof(aux));
-
                     init_pin_ctx(&pin_ctx, G_pin_buffer);
-#ifndef DEBUG_BUILD
-                    if (!is_pin_valid(&pin_ctx)) {
-                        THROW(ERR_INVALID_PIN);
-                    }
-#endif
-                    // Wipe device
-                    os_global_pin_invalidate();
-                    os_perso_wipe();
-                    G_bolos_ux_context.onboarding_kind =
-                        BOLOS_UX_ONBOARDING_NEW_24;
-                    // Generate 32 bytes of random with onboard rng
-                    cx_rng((unsigned char *)G_bolos_ux_context.string_buffer,
-                           HASHSIZE);
-                    // XOR with host-generated 32 bytes random
-                    for (i = 0; i < HASHSIZE; i++) {
-                        G_bolos_ux_context.string_buffer[i] ^=
-                            G_bolos_ux_context.words_buffer[i];
-                    }
-                    // The seed is now in string_buffer, generate the mnemonic
-                    os_memset(G_bolos_ux_context.words_buffer,
-                              0,
-                              sizeof(G_bolos_ux_context.words_buffer));
-                    G_bolos_ux_context.words_buffer_length =
-                        bolos_ux_mnemonic_from_data(
-                            (unsigned char *)G_bolos_ux_context.string_buffer,
-                            SEEDSIZE,
-                            (unsigned char *)G_bolos_ux_context.words_buffer,
-                            sizeof(G_bolos_ux_context.words_buffer));
-                    // Clear the seed
-                    explicit_bzero(G_bolos_ux_context.string_buffer,
-                                   sizeof(G_bolos_ux_context.string_buffer));
-                    // Set seed from mnemonic
-                    os_perso_derive_and_set_seed(
-                        0,
-                        NULL,
-                        0,
-                        NULL,
-                        0,
-                        G_bolos_ux_context.words_buffer,
-                        strlen(G_bolos_ux_context.words_buffer));
-                    // Clear the mnemonic
-                    explicit_bzero(G_bolos_ux_context.words_buffer,
-                                   sizeof(G_bolos_ux_context.words_buffer));
-                    // Set PIN
-                    os_perso_set_pin(
-                        0, GET_PIN(&pin_ctx), GET_PIN_LENGTH(&pin_ctx));
-                    // Finalize onboarding
-                    os_perso_finalize();
-                    os_global_pin_invalidate();
-                    SET_APDU_AT(1, 2);
-                    SET_APDU_AT(2,
-                                os_global_pin_check(GET_PIN(&pin_ctx),
-                                                    GET_PIN_LENGTH(&pin_ctx)));
+                    tx = wipe_device(&onboard_ctx, &pin_ctx);
```

Comment:
This should probably be something more along the lines of `onboard_device` since we not only wipe it, but mainly onboard it.